tests/benchq/algorithms/test_lde_solver.py

```diff
@@ -125,3 +126,21 @@ def test_inverse_blockencoding(beta, epsilon, time):
     assert isinstance(sel_inv, Circuit)
     assert sel_inv.n_qubits == total_qubits
     assert expected_num_of_custom_gates == actual_num_of_custom_gates
+
+
+@pytest.mark.parametrize(
+    "beta, epsilon, time",
+    [
+        (1.3, 1e-3, 0.1),
+        (1.5, 1e-3, 0.1),
+    ],
+)
+def test_matrix_exponentiation(beta, epsilon, time):
+    matrix_norm, dummy_circuit = construct_dummy_cir()
+    sel_inv = inverse_blockencoding(dummy_circuit, matrix_norm, time, beta, epsilon)
+    total_qubits = sel_inv.n_qubits
+
+    matrix_exp = matrix_exponentiation(dummy_circuit, matrix_norm, time, beta, epsilon)
+
+    assert isinstance(matrix_exp, Circuit)
+    assert matrix_exp.n_qubits == total_qubits
```

Comment:
We might want to come up with some better assertion statements rather than just the totol number of qubits. I'll have to think about this later....

Reply:
Would it make sense to check that `matrix_exp.to_unitary()` contains a block that is close to the relevant block of `dummy_circuit.to_unitary()`?